### ISSUE TITLE
Documentation for hosting caja server

### ISSUE THREAD

**Issue by Marchhill** [google/caja#2030]
I am trying to make guest pages able to include external scripts, however this does not work using the "https://caja.appspot.com", however this does not allow external scripts. How can I run my own caja server, is there any documentation?

**Comment by kpreid** [comment]
Yes, the documentation never got written for how to do this properly and it's a bit of a mess overall.

If the external scripts come from a CORS-enabled server, then you don't need any "Caja server" at all, just [the JavaScript files](https://groups.google.com/d/msg/google-caja-discuss/WH1nX6UhNfo/b95tk_Em2ksJ):

> We use ant for our build system, so make sure you have ant installed and then run it in the source tree. All of the JS files will be compiled to:  
>    ant-lib/com/google/caja/plugin/*.js
> 
> Note that this includes files you don't need as well as ones you do. The only ones you need are caja.js and ses-single-frame.js, unless you're making use of standalone components like the HTML sanitizer or SES, or using the Google APIs taming.

Then in your configuration you specify the URI policy as `{fetcher: caja.policy.net.fetcher.USE_XHR, rewriter: caja.policy.net.rewriter.ALL}`.

If at all possible, I recommend this mode of operation.

Otherwise, if you have a Java servlet environment you can load `com.google.caja.service.ProxyServlet`, or (least pleasantly) reimplement the proxy protocol provided by it on your own server platform.

**Comment by Marchhill** [comment]
Thank you for your very quick reply, I have set up my local server with CORS enabled and have changed the URI policy, however I still receive the same error message.

> 
> Uncaught script error: Uncaught Error: not loaded in source: "http://localhost:8000/test.js" at line: -1
> ses-single-frame.js:34340:5

**Comment by kpreid** [comment]
For debugging, let's take this to https://groups.google.com/group/google-caja-discuss instead of the issue tracker. When you post, please include the script you're using to configure Caja and load the code.

**Comment by dgtlmoon** [comment]
@kpreid Hmm this kind of information should go way higher up in the README.md , I spent hours trying to compile the code, and eventually found a docker image which runs very nicely - but I still can't figure out how to use the docker image which exposes an API ( `tmlbl/caja` ) so after all this, you're saying you don't actually need that caja core server, and you can use some JS instead? I'm totally lost

**Comment by dgtlmoon** [comment]
Also related https://github.com/google/caja/pull/1987
